src/util/small_map.h

```diff
@@ -0,0 +1,560 @@
+/*******************************************************************\
```

Comment:
What does "small map" mean? Can you elaborate on this in the commit message itself?